File: 1200/1133C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int n;
    cin >> n;

    vector<int> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];

    sort(a.begin(), a.end());

    int maxi = INT_MIN;
    int l = 0, r = 1;

    while (r < n) 
    {
        if (a[r] - a[l] > 5) 
        {
            maxi = max(maxi, r - l);
            while (l <= r && a[r] - a[l] > 5)
                l++;
        }
        
        r++;
    }
    maxi = max(maxi, r - l);

    cout << maxi << "\n";

    return 0;
}

File: 1200/1133C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        
        int a[] = new int[n];
        for(int i=0;i<n;i++)
            a[i] = sc.nextInt();
        
        Arrays.sort(a);
        
        int maxi = Integer.MIN_VALUE;
        
        int l = 0;
        int r = 1;
        
        while(r < n)
        {
            if(a[r]-a[l] > 5)
            {
                maxi = Math.max(maxi, r-l);
                while(l<=r && a[r]-a[l] > 5)
                    l++;
            }
            
            r++;
        }
        maxi = Math.max(maxi, r-l);
        
        System.out.println(maxi);
    }
}